File: UI/src/app/UriMapper.ts

export const BASE_URI = 'http://localhost:8080/api'

export const AUTH_BASE = BASE_URI + '/auth'
export const LOGIN = AUTH_BASE + '/login'

export const MEASUREMENT_BASE = BASE_URI +  '/measurement'
export const GET_MEASUREMENTS_FOR_DEVICE = MEASUREMENT_BASE + '/get-all-of-device'
export const GET_MEASUREMENTS_FOR_DEVICE_BY_DAY = MEASUREMENT_BASE + '/get-by-day'

export const BASE_USER = BASE_URI +  '/users'
export const GET_ALL_STANDARD_USERS  = BASE_USER + '/get_all_regular'
export const REGISTER_USER = BASE_USER + '/register-user'
export const EDIT_USER = BASE_USER + '/edit'
export const DELETE_USER = BASE_USER + '/delete'

export const ADMIN_BASE = BASE_USER + '/admin'
export const CREATE_ADMIN = ADMIN_BASE + '/create'

export const BASE_DEVICE = BASE_URI + '/device'
export const GET_ALL_DEVICES = BASE_DEVICE + '/get-all'
export const GET_DEVICES_BY_USER = BASE_DEVICE + '/get-all-by-user'
export const GET_DEVICE_BY_ID = BASE_DEVICE + '/get'
export const ASSIGN_OWNER = BASE_DEVICE + '/assign-owner'
export const CREATE_DEVICE = BASE_DEVICE + '/create'
export const EDIT_DEVICE = BASE_DEVICE + '/edit'
export const DELETE_DEVICE = BASE_DEVICE + '/delete'

File: src/main/java/ro/bogdanenergy/energymonitoringsystem/UriMapper.java
package ro.bogdanenergy.energymonitoringsystem;

import ro.bogdanenergy.energymonitoringsystem.model.Device;

public class UriMapper {
    public static final String BASE = "/api";

    public static final String AUTH = BASE +  "/auth";
    public static final String LOGIN = AUTH + "/login";
    public static final String REFRESH_TOKEN = AUTH +  "/refresh-token";
    public static final String CHANGE_PASSWORD = AUTH + "/change-password";


    public static final String ADMIN_BASE = "/admin";
    public static final String REGISTER_ADMIN = ADMIN_BASE + "/create";


    public static final String USER_BASE = BASE +  "/users";
    public static final String CREATE_USER = "/register-user";
    public static final String GET_ALL_USERS = "/get-users";
    public static final String EDIT_USER = USER_BASE + "/edit";
    public static final String DELETE_USER = ADMIN_BASE + "/delete";
    public static final String GET_ALL_STANDARD_USERS = USER_BASE + "/get_all_regular";


    public static final String DEVICE_BASE = BASE + "/device";
    public static final String CREATE_DEVICE = "/create";
    public static final String GET_DEVICE_BY_ID = "/get";
    public static final String GET_ALL_DEVICES = "/get-all";
    public static final String GET_DEVICES_OF_USER = "/get-all-by-user";
    public static final String EDIT_DEVICE = "/edit";
    public static final String ASSIGN_OWNER = "/assign-owner";
    public static final String DELETE_DEVICE = "/delete";
    public static final String GET_RANDOM_ID = "/get-random-id";

    public static final String MEASUREMENT_BASE = BASE + "/measurement";
    public static final String GET_ALL_MEASUREMENTS_OF_DEVICE = "/get-all-of-device";
    public static final String GET_ALL_MEASUREMENTS_OF_DEVICE_BY_DAY = "/get-by-day";
    public static final String GET_ALL_MEASUREMENTS_OF_USER = "/get-all-of-user";
    public static final String GET_MEASUREMENTS_OF_DEVICE_IN_INTERVAL = "/get-measurements-in-interval";
    public static final String GET_MEASUREMENT = "/get-measurement";
    public static final String CREATE_MEASUREMENT = "/create";
    public static final String EDIT_MEASUREMENT = "/edit";
    public static final String DELETE_MEASUREMENT = "/delete";

}
